src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1451,70 +1455,31 @@ private void CheckMemberNameConflicts(DiagnosticBag diagnostics)
                     // second and third categories as follows:
 
                     var conversion = symbol as SourceUserDefinedConversionSymbol;
-                    var method = symbol as SourceMethodSymbol;
+                    // Does this conversion collide with any previously-seen conversion?
                     if ((object)conversion != null)
                     {
-                        // Does this conversion collide *as a conversion* with any previously-seen
-                        // conversion?
-
-                        SourceUserDefinedConversionSymbol previousConversion;
-                        if (conversionsAsConversions.TryGetValue(conversion, out previousConversion))
+                        if (!conversionsBySignature.Add(conversion))
```

Comment:
I believe method signature collisions can be handled by `methodsBySignature` alone. As a result, for consistency, `conversionsAsConversions` was renamed to `conversionsBySignature`.